ext/CEGUI/IrrlichtBaWTexture.cpp

```diff
@@ -127,6 +198,11 @@ void IrrlichtBaWTexture::loadFromMemory(const void* buffer, const CEGUI::Sizef&
         }
     }();
 
+    auto texelOrBlockByteSize = asset::getTexelOrBlockBytesize(newFormat);
+    auto bufferByteSize = buffer_size.d_width * buffer_size.d_height * texelOrBlockByteSize;
+    core::smart_refctd_ptr<asset::ICPUBuffer> cpuPixelBuffer = core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>>>(bufferByteSize, buffer, core::adopt_memory);
+    core::smart_refctd_ptr<video::IGPUBuffer> gpuPixelBuffer = driver->getGPUObjectsFromAssets(&cpuPixelBuffer.get(), &cpuPixelBuffer.get() + 1u)->front();
```

Comment:
create a GPU buffer of the given `bufferByteSize` with `driver->createGPUBufferDeviceLocalOnDedMem` and fill it with `driver->updateBufferRangeViaStagingBuffer`, no need for `getGPUObjectsFromAssets` and creation of CPU buffer

Reply:
adjusted in following commit